pkg/plugins/policies/meshhealthcheck/plugin/v1alpha1/plugin_test.go

```diff
@@ -272,6 +287,162 @@ var _ = Describe("MeshHealthCheck", func() {
 		}),
 	)
 
+	It("should generate correct configuration for MeshExternalService with ZoneEgress", func() {
+		// given
+		rs := core_xds.NewResourceSet()
+		rs.Add(&core_xds.Resource{
+			Name:           "external-default",
+			Origin:         egress.OriginEgress,
+			Resource:       clusters.NewClusterBuilder(envoy_common.APIV3, "external").MustBuild(),
+			ResourceOrigin: backendMeshExternalServiceIdentifier("default"),
+			Protocol:       core_mesh.ProtocolTCP,
+		})
+		rs.Add(&core_xds.Resource{
+			Name:           "external-mesh2",
+			Origin:         egress.OriginEgress,
+			Resource:       clusters.NewClusterBuilder(envoy_common.APIV3, "external").MustBuild(),
+			ResourceOrigin: backendMeshExternalServiceIdentifier("mesh2"),
+			Protocol:       core_mesh.ProtocolTCP,
+		})
+
+		proxy := &core_xds.Proxy{
+			APIVersion: envoy_common.APIV3,
+			ZoneEgressProxy: &core_xds.ZoneEgressProxy{
+				ZoneEgressResource: &core_mesh.ZoneEgressResource{
+					Meta: &test_model.ResourceMeta{Name: "dp1", Mesh: "default"},
+					Spec: &mesh_proto.ZoneEgress{
+						Networking: &mesh_proto.ZoneEgress_Networking{
+							Address: "192.168.0.1",
+							Port:    10002,
+						},
+					},
+				},
+				ZoneIngresses: []*core_mesh.ZoneIngressResource{},
+				MeshResourcesList: []*core_xds.MeshResources{
+					{
+						Mesh: builders.Mesh().WithName("default").WithEnabledMTLSBackend("ca-1").WithBuiltinMTLSBackend("ca-1").Build(),
+						Resources: map[core_model.ResourceType]core_model.ResourceList{
+							meshexternalservice_api.MeshExternalServiceType: &meshexternalservice_api.MeshExternalServiceResourceList{
+								Items: []*meshexternalservice_api.MeshExternalServiceResource{
```

Comment:
please use `builder` package to construct these types in the test https://github.com/kumahq/kuma/blob/45eed3ba79d2765cfd16ea30d70527d5dfd2a611/pkg/test/resources/builders/meshexternalservice_builder.go#L1-L0